File: javascript/Type.js

// 枚举类型
class Type {
    constructor() {
        // auto 是单字符符号的类型
        this.auto = 'auto' // 0
        // this.colon = 'colon' // 1
        // this.comma = 'comma' // 2
        // this.braceLeft = 'braceLeft' // 3
        // this.braceRight = 'barceRight' //4
        this.bracketLeft = 'bracketLeft' //5
        this.bracketRight = 'bracketRight' //6
        this.number = 'number' //7
        this.string = 'string' //8
        // this.token = 'token' //9
        
        this.operator = 'operator'
        this.bool = 'bool'
        this.comment = 'comment'
        this.condition = 'condition'
        this.id = 'identity'
    }
}

module.exports = new Type()

File: python/type.py
from enum import Enum
class Type(Enum):
    auto = 0            # auto 就是 6 个单字符符号的类型
    # colon = 1
    # comma = 2
    # braceLeft = 3
    # braceRight = 4
    bracketLeft = 5
    bracketRight = 6
    number = 7
    string = 8
    # token = 9

    operator = 10
    bool = 12
    comment = 13
    condition = 14
    id = 15
